core-services/src/main/java/org/zalando/nakadi/cache/ChangeSet.java

```diff
@@ -60,9 +61,9 @@ private Date getActualChangeDate(final Change change) {
         for (final Map.Entry<String, List<Change>> newEntry : newChangeMap.entrySet()) {
             // figure out proper dates of the changes
             final List<Change> newChangesSorted = newEntry.getValue().stream()
-                    .sorted(Comparator.comparing(this::getActualChangeDate).reversed())
+                    .sorted(Comparator.comparing(this::getActualChangeDate).thenComparing(Change::getId).reversed())
                     .collect(Collectors.toList());
-            final long newestAge = System.currentTimeMillis() - newChangesSorted.get(0).getOccurredAt().getTime();
+            final long newestAge = System.currentTimeMillis() - getActualChangeDate(newChangesSorted.get(0)).getTime();
```

Comment:
The fix is : 
`newChangesSorted.get(0).getOccurredAt()` -> `getActualChangeDate(newChangesSorted.get(0))`

Reply:
Everything else is to guarantee the same behavior on all the nodes